eras/babbage/impl/src/Cardano/Ledger/Babbage/Collateral.hs

```diff
@@ -36,19 +40,17 @@ isTwoPhaseScriptAddress ::
   Bool
 isTwoPhaseScriptAddress tx utxo = isTwoPhaseScriptAddressFromMap @era (txscripts utxo tx)
 
-collBalance ::
-  forall era.
+collAdaBalance ::
   (BabbageEraTxBody era, TxOut era ~ BabbageTxOut era) =>
   TxBody era ->
-  UTxO era ->
-  Value era
-collBalance txBody (UTxO m) =
+  Map.Map (TxIn (Crypto era)) (TxOut era) ->
```

Comment:
In a few places I see we've shed the constructor and work directly with the `Map`. Would it be clearer if we kept it as `UTxO`? Or maybe there's a reason for using a map that I haven't understood... :thinking:

Reply:
I did it on purpose, because before `callBalance` accepted the whole of `UTxO`, but now it only suppose to accept `eval (txBody ^. collateralInputsTxBodyL) ◁ utxo)`. So I wanted to get a type error instead of it happily accepting either one of those values. Also it is more consistent with Alonzo:
https://github.com/input-output-hk/cardano-ledger/blob/2eafa6e55723c4cd2ef04911674e8f66830661d5/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs#L315

Reply:
great :+1: